discovery-infra/test_infra/helper_classes/kube_helpers.py

```diff
@@ -0,0 +1,376 @@
+            reason = condition.get('reason')
+
+            if reason == 'AgentPlatformState':
+                state = condition.get('message')
+
+            # todo: fix typo after it will be fixed in assisted-service
+            #  (Agnet -> Agent)
+            elif reason == 'AgnetPlatformStateInfo':
+                state_info = condition.get('message')
+
+            if state and state_info:
+                break
+
+        return state, state_info
+
+    def wait_for_state(
+            self,
+            required_state: str,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATE_TIMEOUT
+    ) -> None:
+
+        required_state = required_state.lower()
+
+        def _has_required_state() -> bool:
+            state, _ = self.state(0.5)
+            return state.lower() == required_state
+
+        waiting.wait(
+            _has_required_state,
+            timeout_seconds=timeout,
+            waiting_for=f'cluster {self.ref} state to be {required_state}',
+            expected_exceptions=waiting.exceptions.TimeoutExpired
+        )
+
+
+def deploy_default_secret(
+        name: str,
+        ignore_conflict: bool = True,
+        pull_secret: Optional[str] = None
+) -> Secret:
+
+    if pull_secret is None:
+        pull_secret = os.environ.get('PULL_SECRET', '')
+    secret = Secret(name)
+    try:
+        secret.create(pull_secret)
+    except ApiException as e:
+        if e.reason != 'Conflict' or not ignore_conflict:
+            raise
+    return secret
+
+
+def deploy_default_cluster_deployment(
+        name: str,
+        ignore_conflict: bool = True,
+        base_domain: str = env_variables['base_domain'],
+        platform_params: Optional[dict] = None,
+        install_strategy_params: Optional[dict] = None,
+        secret: Optional[Secret] = None,
+        **kwargs
+) -> ClusterDeployment:
+
+    platform = Platform(**platform_params or {})
+    install_strategy = InstallStrategy(**install_strategy_params or {})
+
+    if secret is None:
+        secret = deploy_default_secret(name, ignore_conflict=ignore_conflict)
+
+    cluster_deployment = ClusterDeployment(name)
+    try:
+        cluster_deployment.create(
+            platform=platform,
+            install_strategy=install_strategy,
+            secret=secret,
+            base_domain=base_domain,
+            **kwargs
+        )
+    except ApiException as e:
+        if e.reason != 'Conflict' or not ignore_conflict:
+            raise
+
+    # wait until cluster will have status (i.e until resource will be
+    # processed in assisted-service).
+    cluster_deployment.status()
+
+    return cluster_deployment
+
+
+def delete_cluster_deployment(
+        cluster_deployment: ClusterDeployment,
+        ignore_not_found: bool = True
+) -> None:
+
+    def _try_to_delete_resource(o):
+        try:
+            o.delete()
+        except ApiException as e:
+            if e.reason != 'Not Found' or not ignore_not_found:
```

Comment:
I think `if not (e.reason == 'Not Found' and ignore_not_found)` is more readable.